extensions/gossip/state/state.go

```diff
@@ -57,7 +57,7 @@ func AddBlockHandler(publisher api.BlockPublisher) {
 }
 
 //NewGossipStateProviderExtension returns new GossipStateProvider Extension implementation
-func NewGossipStateProviderExtension(chainID string, mediator GossipServiceMediator) GossipStateProviderExtension {
+func NewGossipStateProviderExtension(chainID string, mediator GossipServiceMediator, preCommit interface{}) GossipStateProviderExtension {
```

Comment:
What is the purpose of preCommit?

Reply:
To trigger BlockEventer (exposed as AddBlock() is ledger committer), which does current config update and calls blockPublisher.Publish. Right now, it gets called only for committer during storeblock. As a fix, endorser will call this block eventer directly. Similar to cluster changes.